evidence/pool.go

```diff
@@ -96,34 +98,54 @@ func (evpool *Pool) Update(block *types.Block, state sm.State) {
 
 // AddEvidence checks the evidence is valid and adds it to the pool.
 func (evpool *Pool) AddEvidence(evidence types.Evidence) error {
-
-	// check if evidence is already stored
-	if evpool.store.Has(evidence) {
-		return ErrEvidenceAlreadyStored{}
+	state := evpool.State()
```

Comment:
I'm not too sure where exactly but we should do a quick `ValidateBasic()` of the evidence before we get into the heavier processing. Either here or perhaps when we loop through the evList right after checking whether the evidence is already in the database

Reply:
`ValidateBasic` is called in https://github.com/tendermint/tendermint/blob/4647810d8509686e2e3d38d18ac1c51a97d7de32/rpc/core/evidence.go#L15 when evidence is submitted through RPC and https://github.com/tendermint/tendermint/blob/4647810d8509686e2e3d38d18ac1c51a97d7de32/evidence/reactor.go#L73 when evidence is submitted through P2P.